packages/purpose-process/src/services/readModelService.ts

```diff
@@ -158,9 +158,38 @@ async function buildGetPurposesAggregation(
     : {};
 
   const consumersIdsFilter: ReadModelFilter<Purpose> =
-    ReadModelRepository.arrayToFilter(consumersIds, {
-      "data.consumerId": { $in: consumersIds },
-    });
+    consumersIds.length > 0
+      ? {
+          $or: [
+            { "data.consumerId": { $in: consumersIds } },
+            {
+              $and: [
+                {
+                  $or: [
+                    {
+                      "matchingDelegations.data.delegateId": {
+                        $in: consumersIds,
+                      },
+                    },
+                    {
+                      "matchingDelegations.data.delegatorId": {
+                        $in: consumersIds,
+                      },
+                    },
+                  ],
```

Comment:
Since we save the `delegationId` in the purpose, here we should also add a check that the delegation we found is the one in the purpose, something like:

```
 $eq: ["$$delegation.data.id", "$data.delegationId"], 
```